From: Tzafrir Cohen <tzafrir@debian.org>
Subject: Handle a return value that must be handled

The return value of write must be checked. In this case: stop the loop
if the write failed.

--- a/pritest.c
+++ b/pritest.c
@@ -64,11 +64,16 @@ static void do_channel(int fd)
 {
 	/* This is the part that runs on a given channel */
 	char buf[READ_SIZE];
-	int res;
+	int res, res_write;
 	int i=0;
 
 	while ((res = read(fd, buf, READ_SIZE)) > 0 && (i++ < 1000)) {
-		write(fd, buf, res);
+		res_write = write(fd, buf, res);
+		if (res_write == -1) {
+			fprintf(stderr, "--!! %s: Failed write: %d\n",
+					__func__, errno);
+			break;
+		}
 	}
 }